File: src/main/java/com/example/payheurebackend/service/PointageAnalyticsService.java
package com.example.payheurebackend.service;

import com.example.payheurebackend.dto.PointageAnalyticsRequest;
import com.example.payheurebackend.dto.PointageAnalyticsResponse;

/**
 * Statistiques de complétude des pointages (sessions complètes/incomplètes et ratio), tous
 * salariés confondus, agrégées par jour, semaine ou mois.
 */
public interface PointageAnalyticsService {

    /**
     * Reconstitue les sessions de travail de tous les salariés actifs sur la période demandée et
     * les agrège par jour, semaine ou mois selon {@code request.granularite()}.
     *
     * @throws com.example.payheurebackend.exception.InvalidPeriodException si la période est incohérente
     */
    PointageAnalyticsResponse calculer(PointageAnalyticsRequest request);
}

File: PayHeureUI/src/app/services/pointage-analytics.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PointageAnalyticsRequest, PointageAnalyticsResponse } from '../models/PointageAnalytics';
import { API_BASE_URL } from '../config/api.config';

@Injectable({
  providedIn: 'root'
})
export class PointageAnalyticsService {

  private readonly apiUrl = `${API_BASE_URL}/pointages`;

  constructor(private http: HttpClient) {}

  /** Statistiques de complétude des pointages, tous salariés confondus, agrégées par jour/semaine/mois. */
  calculer(request: PointageAnalyticsRequest): Observable<PointageAnalyticsResponse> {
    return this.http.post<PointageAnalyticsResponse>(`${this.apiUrl}/analytics`, request);
  }
}
